cardano-node/src/Cardano/Node/Tracing/Era/Shelley.hs

```diff
@@ -0,0 +1,1038 @@
+
+import           Ouroboros.Consensus.Ledger.SupportsMempool (txId)
+import qualified Ouroboros.Consensus.Ledger.SupportsMempool as SupportsMempool
+import qualified Ouroboros.Consensus.Protocol.Ledger.HotKey as HotKey
+import           Ouroboros.Consensus.Protocol.TPraos (TPraosCannotForge (..))
+import           Ouroboros.Consensus.Shelley.Ledger hiding (TxId)
+import           Ouroboros.Consensus.Shelley.Ledger.Inspect
+import           Ouroboros.Consensus.Util.Condense (condense)
+
+import           Cardano.Protocol.TPraos.BHeader (LastAppliedBlock, labBlockNo)
+import           Cardano.Protocol.TPraos.Rules.OCert
+import           Cardano.Protocol.TPraos.Rules.Overlay
+import           Cardano.Protocol.TPraos.Rules.Updn
+
+
+import qualified Cardano.Ledger.Alonzo as Alonzo
+import           Cardano.Ledger.Alonzo.Rules.Bbody (AlonzoBbodyPredFail)
+import qualified Cardano.Ledger.Alonzo.Rules.Utxo as Alonzo
+import qualified Cardano.Ledger.Alonzo.Rules.Utxos as Alonzo
+import           Cardano.Ledger.Alonzo.Rules.Utxow (AlonzoPredFail (..))
+import qualified Cardano.Ledger.Alonzo.Tx as Alonzo
+import qualified Cardano.Ledger.AuxiliaryData as Core
+import           Cardano.Ledger.BaseTypes (strictMaybeToMaybe)
+import           Cardano.Ledger.Chain
+import qualified Cardano.Ledger.Core as Core
+import qualified Cardano.Ledger.Crypto as Core
+import qualified Cardano.Ledger.Era as Ledger
+import qualified Cardano.Ledger.SafeHash as SafeHash
+import qualified Cardano.Ledger.ShelleyMA.Rules.Utxo as MA
+import qualified Cardano.Ledger.ShelleyMA.Timelocks as MA
+
+-- TODO: this should be exposed via Cardano.Api
+import           Cardano.Ledger.Shelley.API hiding (ShelleyBasedEra)
+
+import           Cardano.Ledger.Shelley.Rules.Bbody
+import           Cardano.Ledger.Shelley.Rules.Deleg
+import           Cardano.Ledger.Shelley.Rules.Delegs
+import           Cardano.Ledger.Shelley.Rules.Delpl
+import           Cardano.Ledger.Shelley.Rules.Epoch
+import           Cardano.Ledger.Shelley.Rules.Ledger
+import           Cardano.Ledger.Shelley.Rules.Ledgers
+import           Cardano.Ledger.Shelley.Rules.Mir
+import           Cardano.Ledger.Shelley.Rules.NewEpoch
+import           Cardano.Ledger.Shelley.Rules.Newpp
+import           Cardano.Ledger.Shelley.Rules.Pool
+import           Cardano.Ledger.Shelley.Rules.PoolReap
+import           Cardano.Ledger.Shelley.Rules.Ppup
+import           Cardano.Ledger.Shelley.Rules.Rupd
+import           Cardano.Ledger.Shelley.Rules.Snap
+import           Cardano.Ledger.Shelley.Rules.Tick
+import           Cardano.Ledger.Shelley.Rules.Upec
+import           Cardano.Ledger.Shelley.Rules.Utxo
+import           Cardano.Ledger.Shelley.Rules.Utxow
+
+import           Cardano.Tracing.OrphanInstances.Shelley ()
+
+{- HLINT ignore "Use :" -}
+
+--
+-- | instances of @LogFormatting@
+--
+-- NOTE: this list is sorted in roughly topological order.
+
+instance (  ToJSON (SupportsMempool.TxId (GenTx (ShelleyBlock era)))
```

Comment:
We usually parameterize by `blk`:
```
instance ( ToJSON (SupportsMempool.TxId (GenTx blk))
         , SupportsMempool.HasTxId (GenTx blk)
         , Condense (GenTx blk)
         ) => LogFormatting (GenTx blk) where
  forMachine dtal tx =
    mkObject $
        ( "txid" .= txId tx )
      : [ "tx"   .= condense tx | dtal == DDetailed ]
 ```

Reply:
Using your code I get the error:
```
src/Cardano/Node/Protocol/Shelley.hs:90:14: error:
    • Overlapping instances for Cardano.Logging.Types.LogFormatting
                                  (Ouroboros.Consensus.Ledger.SupportsMempool.GenTx
                                     (Ouroboros.Consensus.HardFork.Combinator.Basics.HardForkBlock
                                        '[Ouroboros.Consensus.Shelley.Ledger.Block.ShelleyBlock
                                            StandardShelley]))
        arising from a use of ‘SomeConsensusProtocol’

```

Reply:
@Jimbo4350, and we're currently basically mirroring the legacy system in the implementation of this instance -- do we really need to go out of our way with this PR?

Reply:
I should have clarified it was in response to this comment: https://github.com/input-output-hk/cardano-node/pull/3450/files#diff-b5ba7a7995617b90fb25b0295b8148a44bab8dcdd6dae087056b48543a547d60R67

It's not an essential change.